includes/Checker/Abstract_Check_Runner.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Abstract_Check_runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Checker\Check_Runner;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use WordPress\Plugin_Check\Checker\Preparations\Universal_Runtime_Preparation;
+
+/**
+ * Abstract Check Runner class.
+ *
+ * @since n.e.x.t
+ */
+abstract class Abstract_Check_Runner implements Check_Runner {
+
+	/**
+	 * Instance of the Checks class.
+	 *
+	 * @since n.e.x.t
+	 * @var Checks
+	 */
+	protected $checks;
+
+	/**
+	 * Instance of the Check_Context class.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $context;
+
+	/**
+	 * Array of Check instances to run.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	protected $checks_to_run;
+
+	/**
+	 * Determines if the current request is intended for the plugin checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return boolean Returns true if the check is for plugin else false.
+	 */
+	abstract public function is_plugin_check();
+
+	/**
+	 * Setup the check context, checks and preparations based on the request.
+	 *
+	 * @since n.e.x.t
+	 */
+	abstract protected function setup_checks();
```

Comment:
Shouldn't this method be called somewhere in here (e.g. at the beginning of `run()`)? Otherwise, we shouldn't require an abstract method if calling that method as well would be responsibility of the child class.

Reply:
@felixarntz this was missed and meant to be called in the `prepare` method. I will update accordingly.

Reply:
@felixarntz after reviewing your other comments I've gone with additional methods to return the specific class instances we need for preparing and running checks. I've updated the comments for each to explain what they should create and return.